src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -86,19 +86,27 @@ public function copy($originFile, $targetFile, $overwriteNewerFiles = false)
     /**
      * Creates a directory recursively.
      *
-     * @param string|iterable $dirs The directory path
-     * @param int             $mode The directory mode
+     * @param string|iterable $dirs    The directory path
+     * @param int             $mode    The directory mode
+     * @param resource|null   $context
      *
      * @throws IOException On any directory creation failure
      */
-    public function mkdir($dirs, $mode = 0777)
+    public function mkdir($dirs, $mode = 0777/*, $context = null*/)
```

Comment:
not aiming to support this in the Filesystem component per se, but for Stream i think it's required and will be tested as such.

The things is we need this context optionally for all related function calls.. so maybe a stateful approach is better: `Filesystem::inContext($context): self;`